pkg/scheduler/internal/heap/heap.go

```diff
@@ -81,28 +80,28 @@ func (h *data) Less(i, j int) bool {
 }
 
 // Len returns the number of items in the Heap.
-func (h *data) Len() int { return len(h.queue) }
+func (h *data[T]) Len() int { return len(h.queue) }
 
 // Swap implements swapping of two elements in the heap. This is a part of standard
 // heap interface and should never be called directly.
-func (h *data) Swap(i, j int) {
+func (h *data[T]) Swap(i, j int) {
 	h.queue[i], h.queue[j] = h.queue[j], h.queue[i]
 	item := h.items[h.queue[i]]
 	item.index = i
 	item = h.items[h.queue[j]]
 	item.index = j
 }
 
-// Push is supposed to be called by heap.Push only.
-func (h *data) Push(kv interface{}) {
-	keyValue := kv.(*itemKeyValue)
+// Push is supposed to be called by container/heap.Push only.
+func (h *data[T]) Push(kv interface{}) {
```

Comment:
`Push(kv *itemKeyValue[T])` would be better? (type check would work at compiling)

Reply:
> supposed to be called by container/heap.Push only.

Okay, I missed this. 
So, Push and Pop use `interface()` to satisfy the interface of container/heap?

Reply:
Looks like Yes, and Golang has no generics support around container/heap yet. Okay, nvm.

Reply:
Exactly, `Push` and `Pop` have to use `interface{}`